agent/tcs/client/client.go

```diff
@@ -137,9 +142,14 @@ func (cs *clientServer) signRequest(payload []byte) []byte {
 func (cs *clientServer) Close() error {
 	if cs.publishTicker != nil {
 		cs.publishTicker.Stop()
-		cs.endPublish <- struct{}{}
+	}
+	if cs.publishHealthTicker != nil {
+		cs.publishHealthTicker.Stop()
 	}
 
+	if cs.cancel != nil {
```

Comment:
It's awesome that we're using contexts for this. However, i think we should get rid of this check. If needed we can populate this in our test files. But, checking `cancelFunc` is nil before invoking it doesn't seem like a clean way to do that imo